caffe2/operators/softmax_op.cc

```diff
@@ -13,18 +13,19 @@ bool SoftmaxOp<float, CPUContext>::RunOnDevice() {
   const int D = X.size_from_dim(canonical_axis);
   auto* Y = Output(0, X.sizes(), at::dtype<float>());
   float* Ydata = Y->template mutable_data<float>();
+  // ReinitializeTensor itself has the effect of caching, so there is no need to check for numel of Tensor
```

Comment:
This is probably correct but note that there might be a slight behavior change here since ReinitializeTensor also caches based on device, not only size.

Reply:
since this is a CPU op I think it should be fine.